### ISSUE TITLE
[RPC] Refactor `torrent-get` to remove duplicate fields

### ISSUE THREAD

**Issue by tearfur** [transmission/transmission#6670]
## Problem

For RPC `torrent-get`, the `fileStats` arg's functionality overlaps with the args `wanted`, `priorities` and `files`.

This is bug-prone, and it is very telling from the fact in that in Transmission's long history, we have ***never*** gotten the docs and code for `torrent-get`'s `wanted` and `fileStats`'s `wanted` right.

Even now, on one hand, `torrent-get`'s `wanted` is specified as an array of Booleans in the docs, while it is actually an array of 0/1 in code. On the other hand, `fileStats`'s `wanted` is specified as an array of 0/1 in the docs, while it is actually an array of Booleans in code.

## Proposed Solution

I suggest we drop `fileStats`, doing so seems to have the least impact on the Qt app and WebUI.

### Impact to Qt app

I've identified a starting point to conform the Qt app to my solution. These should be enough to serve as a headstart for getting the job done.

- Delete this line.
   https://github.com/transmission/transmission/blob/52d1c1f34b29a887717afdfe86e6cfea69896af7/qt/Torrent.cc#L195
- Delete `TR_KEY_fileStats` and add `TR_KEY_files`, `TR_KEY_priorities` and `TR_KEY_wanted` to this array.
   https://github.com/transmission/transmission/blob/52d1c1f34b29a887717afdfe86e6cfea69896af7/qt/Session.cc#L581-L582

### Impact to WebUI

The changes needed should be similar to the Qt app, since the code structure is similar.

https://github.com/transmission/transmission/blob/52d1c1f34b29a887717afdfe86e6cfea69896af7/web/src/torrent.js#L633-L639

### Which application of Transmission?

None

### Which version of Transmission?

52d1c1f34b29a887717afdfe86e6cfea69896af7

**Comment by tearfur** [comment]
An example for the inconsistencies for `wanted`:

```console
$ curl -s -X POST --data '{"method":"torrent-get","arguments":{"ids":1,"fields":["fileStats","wanted"]}}' --header 'X-Transmission-Session-Id: N6O0OjjW1stJlwH6v3wlRu5sfgMTXCP8s60Bo0Zxwu6gCWwR' localhost:9093/transmission/rpc | python -m json.tool
{
    "arguments": {
        "torrents": [
            {
                "fileStats": [
                    {
                        "bytesCompleted": 2374133760,
                        "priority": 0,
                        "wanted": true
                    },
                    {
                        "bytesCompleted": 1680,
                        "priority": 0,
                        "wanted": true
                    }
                ],
                "wanted": [
                    1,
                    1
                ]
            }
        ]
    },
    "result": "success"
}
```

**Comment by killemov** [comment]
Hmmm ... my suggestion, see #6630, was to drop the less structured `*wanted` and `priority*` arrays in favor of the better structured `fileStats` structure, even for `torrent-set`. (Maybe rename the structure for `torrent-set?`) You could even merge the fields `priority` and `wanted` into `priority` with values of and enumeration like `[ NONE, LOW, NORMAL, HIGH ]` or `[ null, -1, 0 ,1 ]`. In [Shift](https://github.com/killemov/Shift) I have already merged them in a single widget:

![file_menu](https://github.com/transmission/transmission/assets/932370/c28171f1-abde-4603-b8d1-b8170dc457d9)

Add an optional `id` or `index` field to the `fileStats` structure when the received/sent array is not complete. This generally saves bandwidth and processing when only getting or setting data for a few files out of 10k+.

So a `torrent-get` example would be:
```Javascript
{"method":"torrent-get","arguments":{"ids":'recently-active',"fields":["id","fileStats"]}}

{
  "arguments": {
    "torrents": [
      {
        "id": 6,
        "fileStats": [
          {
            "id": 7,
            "bytesCompleted": 2374133760,
          }
        ]
      }
    ],
    "result": "success"
  }
}
```
And a `torrent-set` example would be:
```Javascript
{
  "method": "torrent-set",
  "arguments": {
    "ids": [ 6 ],
    "fileStats": [
      {
        "id": 8,
        "priority": "high"
      }
    ]
  }
}
``` 

You could even scrap the method `torrent-rename-path` and integrate that into `torrent-set`:
```Javascript
{
  "method": "torrent-set",
  "arguments": {
    "ids": [ 6 ],
    "fileStats": [
      {
        "id": 8,
        "path": "pathname/filename10"
        "name": "filename8"
      }
    ]
  }
}

**Comment by killemov** [comment]
I would even go so far to suggest merging it all into the `files` array and allow for sub-field querying. So instead of requesting `fileStats` you could request `files.bytesCompleted`, which is the only field that is truly dynamic and not user modifiable.

**Comment by tearfur** [comment]
> was to drop the less structured `*wanted` and `priority*` arrays in favor of the better structured `fileStats` structure

I get it, you like related arguments being grouped. But I imagine the backend code is going to get exponenially more messy as we add more nested stuff, not a fan. Handling structs/objects in C/C++ is just not as flexible as JavaScript.

> ou could even merge the fields `priority` and `wanted` into `priority` with values of and enumeration like `[ NONE, LOW, NORMAL, HIGH ]` or `[ null, -1, 0, 1 ]`

*This* I can get behind, good idea. Can't think of any downsides yet.

**Comment by killemov** [comment]
I agree that implementing it in a single `files` structure may be more complex on the daemon side, but not exponentially. Expanding the `files` structure to contain all fields and then adding sub-field querying wouldn't really be that much harder than handling every sub-field in basically it's own list. The hard part would be scanning the fields for a `.` and then fetch the sub-field or fetch a part of the structure instead of the entire structure. (Complete array of file)

And ... the goal here ... you get a way better structured RPC. (And the RPC does not have to expose the underlying structures per se.)
